File: doc/tutoNLP/positions_spans_marks/l5_span_around_array_write.cpp
// Difficulty: Level 5
// Request: target the instruction span around the first write to d_partial_sums

void f(float* d_partial_sums) {
  d_partial_sums[0] = 0.0f;
  d_partial_sums[1] = 1.0f;
}

File: doc/tutoNLP/positions_spans_marks/l5_span_around_array_write.ml
open Optitrust
open Prelude

(* Difficulty: Level 5
   Request: target the instruction span around the first write to d_partial_sums
   Expected target: [tSpanAround [occFirst; cArrayWrite "d_partial_sums"]]
   Accepted variant: [tSpan [tBefore; occFirst; cArrayWrite "d_partial_sums"] [tAfter; occFirst; cArrayWrite "d_partial_sums"]] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [tSpanAround [occFirst; cArrayWrite "d_partial_sums"]];
)
